src/lustre/lustrePath.ml

```diff
@@ -955,9 +1141,18 @@ let pp_print_stream_section_pt ident_width val_width sect ppf = function
       (pp_print_list (pp_print_stream_pt ident_width val_width) "@,") 
       l
 
-
 (* For modes *)
-let pp_print_modes_section_pt ident_width val_width mode_ident ppf = function 
+let pp_print_modes_section_pt full_contract ident_width val_width mode_ident ppf = function 
+| (active_modes, contract_info) ->
+if full_contract then match contract_info with
+  | (modes, assumed, guarunteed) ->
+  Format.fprintf 
+    ppf 
+    "%a%a%a"
+    (pp_print_stream_section_pt ident_width val_width "Assumptions") (List.map (function (name, stream_type, mode_trace) -> (name, stream_type, List.map (function (tr_val) -> (Some tr_val)) mode_trace) ) assumed)
+    (pp_print_stream_section_pt ident_width val_width "Guarantees") (List.map (function (name, stream_type, mode_trace) -> (name, stream_type, List.map (function (tr_val) -> (Some tr_val)) mode_trace) ) guarunteed)
+    (pp_print_stream_section_pt ident_width val_width "Modes") (List.map (function (name, stream_type, mode_trace) -> (name, stream_type, List.map (function (tr_val) -> (Some tr_val)) mode_trace) ) modes)
```

Comment:
These lines are very long. Define the lists computed with `map` separately and use them here.